### PR TITLE
Adding certificate validation tool.

### PR DESCRIPTION
**Pull request by CIPop**
Provisioning Service certificate validation tool for .Net.
I wasn't able to find a BouncyCastle crypto lib released for NetStandard2.0 (@DMattoon, do you know of anything that we could use instead?)

### CODE REVIEW

File: provisioning/service/samples/GroupCertificateVerificationSample/VerificationCertificateGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                if (key != null && key.Key.IsPrivate)
+                {
+                    privateParameter = key.Key as ECPrivateKeyParameters;
+                }
+            }
+
+            return privateParameter;
+        }
+
+        private static X509Certificate2 GenerateAndSignCertificate(
```
Comment (simonporter) [inline provisioning/service/samples/GroupCertificateVerificationSample/VerificationCertificateGenerator.cs]: A new requirement I know but we should actually break this into generating a CSR and consuming it. This is how it really should happen and makes this extensible to using an actual CSR service.

File: provisioning/service/samples/GroupCertificateVerificationSample/GroupCertificateVerificationSample.csproj [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="BouncyCastle.Crypto.dll" Version="1.8.1" />
```
Comment (alextolp) [inline provisioning/service/samples/GroupCertificateVerificationSample/GroupCertificateVerificationSample.csproj]: >Include [](start = 22, length = 7)

nit: not sure if we should use the one you listed or this one: [https://www.nuget.org/packages/BouncyCastle/](https://www.nuget.org/packages/BouncyCastle/)

the owner of this pkg is listed as Bouncy Castle Project Contributors

Comment (CIPop) [reply provisioning/service/samples/GroupCertificateVerificationSample/GroupCertificateVerificationSample.csproj]: Thanks @alextolp: I was actually looking into this right now. Changed to the official one.

File: provisioning/service/samples/GroupCertificateVerificationSample/VerificationCertificateGenerator.cs (hunk 1 above)
Comment (CIPop) [reply provisioning/service/samples/GroupCertificateVerificationSample/VerificationCertificateGenerator.cs]: While that may be the way a real CA or Intermediate CA would do it, this is just providing a way for the developer to bootstrap their development process. 

On Windows, the methods I'm aware of are via a [wizard](https://technet.microsoft.com/en-us/library/cc754490(v=ws.11).aspx) or via [certreq](https://technet.microsoft.com/en-us/library/dn296456(v=ws.11).aspx).

Comment (simonporter) [reply provisioning/service/samples/GroupCertificateVerificationSample/VerificationCertificateGenerator.cs]: This serves the dev bootstrap scenario well, we can produce CSR based samples as a more formal way of doing this.

File: provisioning/service/samples/GroupCertificateVerificationSample/Program.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+using System;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.Azure.Devices.Provisioning.Samples
+{
+    public class Program
+    {
+        private const string VerificationCertificatePath = "verificationCertificate.cer";
```
Comment (jasmineymlo) [inline provisioning/service/samples/GroupCertificateVerificationSample/Program.cs]: Please add <> in the verification cert path to indicate that users have to replace with their own values.  Our other samples have something like:
private const string DeviceConnectionString = "<replace>";

**Comment by CIPop** [comment]
Thanks for reviewing @alextolp @jasmineymlo @simonporter !

File: provisioning/service/samples/GroupCertificateVerificationSample/VerificationCertificateGenerator.cs (hunk 1 above)
Comment (CIPop) [reply provisioning/service/samples/GroupCertificateVerificationSample/VerificationCertificateGenerator.cs]: I'll start an internal thread to have this added to the documentation on OSs that we support.

Comment (CIPop) [reply provisioning/service/samples/GroupCertificateVerificationSample/VerificationCertificateGenerator.cs]: @yzhong94 PTAL

**Pull request merged**